crates/parser/src/stake_pool_program/instruction_parser.rs

```diff
@@ -0,0 +1,729 @@
+use borsh::BorshDeserialize;
+use spl_stake_pool::instruction::StakePoolInstruction;
+
+use super::instruction_helpers::{
+    AddValidatorToPoolAccounts, AddValidatorToPoolData, CleanupRemovedValidatorEntriesAccounts,
+    CreateTokenMetadataAccounts, CreateTokenMetadataData, DecreaseAdditionalValidatorStakeAccounts,
+    DecreaseAdditionalValidatorStakeData, DecreaseValidatorStakeAccounts,
+    DecreaseValidatorStakeData, DecreaseValidatorStakeWithReserveAccounts,
+    DecreaseValidatorStakeWithReserveData, DepositSolAccounts, DepositSolData,
+    DepositSolWithSlippageAccounts, DepositSolWithSlippageData, DepositStakeAccounts,
+    DepositStakeWithSlippageAccounts, DepositStakeWithSlippageData,
+    IncreaseAdditionalValidatorStakeAccounts, IncreaseAdditionalValidatorStakeData,
+    IncreaseValidatorStakeAccounts, IncreaseValidatorStakeData, InitializeAccounts, InitializeData,
+    RemoveValidatorFromPoolAccounts, SetFeeAccounts, SetFeeData, SetFundingAuthorityAccounts,
+    SetFundingAuthorityData, SetManagerAccounts, SetPreferredValidatorAccounts,
+    SetPreferredValidatorData, SetStakerAccounts, StakePoolProgramIx,
+    UpdateStakePoolBalanceAccounts, UpdateTokenMetadataAccounts, UpdateTokenMetadataData,
+    UpdateValidatorListBalanceAccounts, UpdateValidatorListBalanceData, WithdrawSolAccounts,
+    WithdrawSolData, WithdrawSolWithSlippageAccounts, WithdrawSolWithSlippageData,
+    WithdrawStakeAccounts, WithdrawStakeData, WithdrawStakeWithSlippageAccounts,
+    WithdrawStakeWithSlippageData,
+};
+
+#[derive(Debug, Copy, Clone)]
+pub struct InstructionParser;
+
+impl yellowstone_vixen_core::Parser for InstructionParser {
+    type Input = yellowstone_vixen_core::instruction::InstructionUpdate;
+    type Output = StakePoolProgramIx;
+
+    fn id(&self) -> std::borrow::Cow<str> {
+        "StakePool::InstructionParser".into()
+    }
+
+    fn prefilter(&self) -> yellowstone_vixen_core::Prefilter {
+        yellowstone_vixen_core::Prefilter::builder()
+            .transaction_accounts([spl_stake_pool::id()])
+            .build()
+            .unwrap()
+    }
+
+    async fn parse(
+        &self,
+        ix_update: &yellowstone_vixen_core::instruction::InstructionUpdate,
+    ) -> yellowstone_vixen_core::ParseResult<Self::Output> {
+        if ix_update.program.equals_ref(spl_stake_pool::id()) {
+            InstructionParser::parse_impl(ix_update)
+        } else {
+            Err(yellowstone_vixen_core::ParseError::Filtered)
+        }
+    }
+}
+
+impl yellowstone_vixen_core::ProgramParser for InstructionParser {
+    #[inline]
+    fn program_id(&self) -> yellowstone_vixen_core::Pubkey {
+        spl_stake_pool::id().to_bytes().into()
+    }
+}
+
+impl InstructionParser {
+    pub(crate) fn parse_impl(
+        ix: &yellowstone_vixen_core::instruction::InstructionUpdate,
+    ) -> yellowstone_vixen_core::ParseResult<StakePoolProgramIx> {
+        let ix_type = StakePoolInstruction::deserialize(&mut ix.data.as_slice()).unwrap();
```

Comment:
Similar comment here, we should be using `try_from_slice` as they are in the program (https://github.com/solana-program/stake-pool/blob/main/program/src/processor.rs#L3440), this adds the buffer lenght check on top of deserialize. Also if you can please handle the error instead of using unwrap this function should return a Result and just use `?` here

```suggestion
        let ix_type = StakePoolInstruction::try_from_slice(ix.data.as_slice()).unwrap();
```

Reply:
Thanks, addresses [3080f03](https://github.com/rpcpool/yellowstone-vixen/pull/87/commits/3080f034b0d3f2daf3a9e0e61edf8e77181e1b1d)